From: Branden Robinson <branden@debian.org>
Subject: downgrade an error to info message

During normal xdm shutdown, killed local X servers
can be zombies; this is not an error.

Forward-ported by David Nusinow, Eugene Konev and Julien Cristau

Index: xdm/dm.c
===================================================================
--- xdm.orig/dm.c
+++ xdm/dm.c
@@ -537,8 +544,17 @@
 			d->startTries, d->startAttempts);
 		if (d->displayType.origin == FromXDMCP ||
 		    d->status == zombie ||
-		    ++d->startTries >= d->startAttempts) {
-		    LogError ("Display %s is being disabled\n", d->name);
+		    ++d->startTries >= d->startAttempts)
+		{
+		    /*
+		     * During normal xdm shutdown, killed local X servers
+		     * can be zombies; this is not an error.
+		     */
+		    if (d->status == zombie &&
+			(d->startTries < d->startAttempts))
+			LogInfo ("display %s is being disabled\n", d->name);
+		    else
+			LogError ("display %s is being disabled\n", d->name);
 		    StopDisplay(d);
 		} else
 		    RestartDisplay (d, TRUE);
